src/backend/gl/src/device.rs

```diff
@@ -205,7 +208,9 @@ impl d::Device<B> for Device {
     fn allocate_memory(
         &self, _mem_type: c::MemoryTypeId, _size: u64,
     ) -> Result<n::Memory, d::OutOfMemory> {
-        unimplemented!()
+        Ok(n::Memory {
+            properties: memory::Properties::CPU_VISIBLE | memory::Properties::CPU_CACHED
```

Comment:
Is this a `TODO` ?

Reply:
We should at least return flags which correspond to `mem_type` (from `PhysicalDevice::memory_properties`). Although I'm not sure how we plan to handle memory allocations in the GL backend in the long term for persistent mapping.